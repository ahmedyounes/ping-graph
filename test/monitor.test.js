/* @flow */
/* eslint-env jest */

const Monitor = require('../lib/monitor')
const Readable = require('stream').Readable
const concat = require('concat-stream')

describe('Monitor', () => {
  it('works', () => {
    return new Promise((resolve, reject) => {
      const stream = makeStream()

      stream.pipe(Monitor()).pipe(concat(data => {
        expect(data).toMatchSnapshot()
        resolve()
      }))

      // $FlowFixMe
      stream.push({ value: 80 })
      // $FlowFixMe
      stream.push({ value: 90 })
      // $FlowFixMe
      stream.push({ value: 100 })
      stream.push(null)
    })
  })
})

function makeStream () {
  return new Readable({ objectMode: true, read () {} })
}
