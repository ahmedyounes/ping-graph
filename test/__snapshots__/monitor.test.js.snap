// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monitor works 1`] = `
Array [
  Object {
    "state": Array [
      80,
    ],
    "value": 80,
  },
  Object {
    "state": Array [
      90,
      80,
    ],
    "value": 90,
  },
  Object {
    "state": Array [
      100,
      90,
      80,
    ],
    "value": 100,
  },
]
`;
